swift/extractor/infra/TargetFile.cpp

```diff
@@ -36,31 +41,49 @@ std::string initPath(std::string_view target, std::string_view dir) {
 TargetFile::TargetFile(std::string_view target,
                        std::string_view targetDir,
                        std::string_view workingDir)
-    : workingPath{initPath(target, workingDir)}, targetPath{initPath(target, targetDir)} {
+    : workingPath{initPath(target, workingDir)}, targetPath{initPath(target, targetDir)} {}
+
+bool TargetFile::init() {
   errno = 0;
   // since C++17 "x" mode opens with O_EXCL (fails if file already exists)
   if (auto f = std::fopen(targetPath.c_str(), "wx")) {
     std::fclose(f);
     out.open(workingPath);
     checkOutput("open file for writing");
-  } else {
-    if (errno != EEXIST) {
-      error("open file for writing", targetPath);
-    }
-    // else we just lost the race, do nothing (good() will return false to signal this)
+    return true;
   }
+  if (errno != EEXIST) {
+    error("open file for writing", targetPath);
+  }
+  // else we just lost the race
+  return false;
+}
+
+std::optional<TargetFile> TargetFile::create(std::string_view target,
+                                             std::string_view targetDir,
+                                             std::string_view workingDir) {
+  TargetFile ret{target, targetDir, workingDir};
+  if (ret.init()) return {std::move(ret)};
```

Comment:
Nit: We can just `return ret;` - the braces are redundant, and named return values are automatically moved from.